### ISSUE TITLE
Memory free tests are flaky

### ISSUE THREAD

**Issue by guynir42** [c3-time-domain/SeeChange#263]
I often get failures in the various `test_free` tests that verify memory being freed. 
In all the cases I've seen, the memory is exactly unchanged before and after freeing. 
I think this may have something to do with a race condition (i.e., the garbage collector didn't have time to do its work). Maybe adding a short sleep after each statement would make it happen. 

It could be something else, as garbage collection is sort of voodoo, but this is worth a try.

**Comment by rknop** [comment]
Agreed it's worth a try, but I fear that the voodoo factor is going to be pretty big.  It probably has various algorithms and heuristics as to when it's worth really freeing the memory vs. caching it aside for future python allocations.  Unless there's a way to tell the garbage collector "release all non-allocated memory back to the system" (not just "run garbage collection"), I predict that this is going to be a very frustrating problem.

**Comment by guynir42** [comment]
I have marked all these tests as "skip" because it is slowing down the development. But we should think more on this, and maybe read more about the garbage collection process and figure out something more robust.
